tools/ratbagctl.test.in

```diff
@@ -0,0 +1,407 @@
+        self.assertEqual(r, "LED: 1 type: logo, mode: cycle, rate: 3, brightness: 40")
+        r = self.launch_good_test("led 1 set brightness 100 test_device")
+        self.assertEqual(r, '')
+        r = self.launch_good_test("led 1 get test_device")
+        self.assertEqual(r, "LED: 1 type: logo, mode: cycle, rate: 3, brightness: 100")
+        self.launch_fail_test("led 1 set brightness this_is_not_an_int test_device")
+        self.launch_fail_test("led 1 set brightness test_device")
+        self.launch_fail_test("led 1 set brightness 100 test_device X")
+        self.launch_fail_test("led 1 set brightness 100 X test_device")
+        self.launch_fail_test("led 1 set brightness this_is_not_an_int test_device")
+        self.launch_fail_test("led 1 set brightness 256 test_device")
+        self.launch_fail_test("led 1 set brightness -10 test_device")
+
+    def test_led_n_set_combination(self):
+        self.setProfile(2)
+        r = self.launch_good_test("led 0 get test_device")
+        self.assertEqual(r, "LED: 0 type: logo, mode: on, color: ff0000")
+        r = self.launch_good_test("led 0 set mode cycle brightness 100 test_device")
+        r = self.launch_good_test("led 0 get test_device")
+        self.assertEqual(r, "LED: 0 type: logo, mode: cycle, rate: 1, brightness: 100")
+        r = self.launch_good_test("led 0 set mode cycle rate 100 brightness 255 test_device")
+        r = self.launch_good_test("led 0 get test_device")
+        self.assertEqual(r, "LED: 0 type: logo, mode: cycle, rate: 100, brightness: 255")
+        r = self.launch_good_test("led 0 set mode on color 00ff00 test_device")
+        r = self.launch_good_test("led 0 get test_device")
+        self.assertEqual(r, "LED: 0 type: logo, mode: on, color: 00ff00")
+        r = self.launch_good_test("led 0 set rate 100 brightness 200 mode cycle test_device")
+        r = self.launch_good_test("led 0 get test_device")
+        self.assertEqual(r, "LED: 0 type: logo, mode: cycle, rate: 100, brightness: 200")
+        r = self.launch_good_test("led 0 set rate 100 brightness 200 mode cycle brightness 100 test_device")
+        r = self.launch_good_test("led 0 get test_device")
+        self.assertEqual(r, "LED: 0 type: logo, mode: cycle, rate: 100, brightness: 100")
+        self.launch_fail_test("led 0 set brightness 10 mode bicycle test_device")
+
+    def test_prefix_led(self):
+        self.setProfile(2)
+        r = self.launch_good_test("Profile 1 Led 1 get test_device")
+        self.assertEqual(r, "LED: 1 type: logo, mode: off")
+        self.launch_fail_test("Profile 1 Profile 2 Led 1 get test_device")
+        self.launch_fail_test("Profile 1 Resolution 2 Led 1 get test_device")
+
+def setUpModule():
+    global ratbagd_process
+    # first copy the policy for the ratbagd daemon to be allowed to run
+    shutil.copy(os.path.join(@MESON_BUILD_ROOT@, 'org.freedesktop.ratbag_devel1.conf'),
+                '/etc/dbus-1/system.d/')
```

Comment:
shutil takes a file as second arg as well, I really think we should have a const here for the path that we pass into os.unlink() below as well.